src/app/services/student-round.service.ts

```diff
@@ -145,6 +145,45 @@ export class StudentRoundService {
     shareReplay(1),
   );
 
+  interactions$: Observable<Interaction[]> =
+  combineLatest([this.sessionService.currentRoundPath$, this.authService.currentUser$]).pipe(
+    switchMap(([path, user]) =>
+      path && user
+        ? this.firebaseService.getStudentInteractions(path, user.uid)
+        : of([])),
+    distinctUntilChanged()
+  );
+
+  totalPollen$: Observable<Interaction[]> = this.interactions$.pipe(
+    map(interactions =>
+      interactions.filter(interaction => interaction.barcodeValue >= 1 && interaction.barcodeValue <= 16)
+    ),
+    distinctUntilChanged(),
+  );
+
+  currentBeePollen$: Observable<number> =
+  this.interactions$.pipe(
+    map(interactions => {
+        let currentPollen = 0;
```

Comment:
Would you mind adding a little comment here? (To an observer, it's not immediately obvious that this depends on the order the interactions are sorted in :slightly_smiling_face:)